### PR TITLE
feat(jtk): detect attachment MIME type and add --filename override for #465

### LINKED ISSUE #465: jtk: add file attachment / image embed support for issues and comments
### Summary

`jtk` has no way to attach a file (e.g. a screenshot) to an issue, or to embed an image in a comment or description. The `comments` and `issues` command groups are text-only, so adding visual evidence to a ticket requires dropping out of the CLI and using the web UI.

### Use case

A common workflow is attaching before/after screenshots (or other image evidence) to a ticket as part of documenting a change. Today that must be done manually in the browser, because `jtk` can only add text comments and update text fields.

### Request

Add attachment/embed support, for example:

- `jtk issues attach <ISSUE-KEY> <path> [<path>...]` — upload one or more files to an issue's attachments.
- Optionally, let an uploaded attachment be referenced inline in a comment or description so images render (Jira's `!filename!` media syntax), so `jtk comments add` / `jtk issues update` can embed images.

Jira Cloud exposes this via `POST /rest/api/3/issue/{key}/attachments` (multipart body, `X-Atlassian-Token: no-check` header); the CLI already authenticates against this API for other commands.

### Notes

- Should reuse the existing auth/keyring flow — no new credential handling.
- A `--filename` override and MIME-type detection would be nice-to-haves.

### PR DESCRIPTION
**Pull request by piekstra**
Closes #465.

## What the issue asked for vs. what already exists

The bulk of #465 — *"attach a file to an issue"* — **already ships today** as `jtk attachments add <issue-key> --file <path>` (list/add/get/delete under the `attachments` group; API in place since #50, registered in `main.go`, documented in the README). So this PR does **not** re-add attach support.

The issue proposed the command as `jtk issues attach …`, which is the likely reason it read as missing: attach lives in the top-level `attachments` group, not under `issues`, so `jtk issues --help` doesn't surface it. That's a findability gap, not a docs-accuracy gap. I deliberately did **not** add a duplicate `jtk issues attach` alias — the repo's `GUARDRAILS.md` defines `attachments`/`add` as the canonical resource/verb, and duplicating it under `issues` would cut against that design language.

What this PR does implement are the two concrete gaps the issue lists as nice-to-haves, which are the only parts of the existing feature that were genuinely absent.

## Changes

- **MIME-type detection on upload.** The multipart part's `Content-Type` is now derived from the file extension (`mime.TypeByExtension`), falling back to `application/octet-stream`.
- **`--filename` override** on `jtk attachments add` — store the attachment under a chosen name instead of the file's base name. Valid with a single `--file` only (errors clearly otherwise).
- **Test coverage** for `AddAttachment`, which previously had none — asserts the wire filename, the detected `Content-Type`, the octet-stream fallback, and the empty-arg guards.

## Why

The motivating use case in the issue is attaching screenshots. Before this change every upload was sent as `application/octet-stream`, so Jira didn't recognize a `.png`/`.pdf` as an image/document — no thumbnail, no inline preview, it renders as a generic download. Detecting the MIME type from the extension fixes that. `--filename` covers the common case of uploading a temp file with an ugly generated name (`/tmp/out-8f8a7b.png`) but wanting it stored as something meaningful (`before.png`).

## Deliberately deferred

The issue's *optional* ask — referencing an uploaded attachment inline in a comment/description via Jira's `!filename!` media syntax — is not included. jtk renders comments/descriptions as ADF (v3 API), and the shared `adf` package has no media-node support; wiring inline embeds would need new shared node types plus a media-services UUID the v3 attachment API doesn't return, and it can't be verified without a live Jira. That's a larger, separately-scoped change and is left for a follow-up.

## Testing

- `go test -race ./tools/jtk/...` — pass
- `golangci-lint run` (tools/jtk) — 0 issues
- `go mod tidy` — clean; cross-compile release guard (linux/windows) — pass
- Verified on the built binary: `--filename` appears in `attachments add --help`; the single-`--file` validation errors with exit 1 before auth is attempted.

### CODE REVIEW

File: tools/jtk/api/attachments.go [hunk 1]
```diff
@@ -6,14 +6,22 @@ import (
...
 	"path/filepath"
+	"strings"
 
 	sharederrors "github.com/open-cli-collective/atlassian-go/errors"
 )
 
+// quoteEscaper mirrors the escaping mime/multipart applies to Content-Disposition
+// parameters, so an override filename containing a quote or backslash cannot break
+// out of the header.
+var quoteEscaper = strings.NewReplacer("\\", "\\\\", `"`, "\\\"")
```
Comment (piekstra-dev) [inline tools/jtk/api/attachments.go:23]: `quoteEscaper` and its use in `attachmentPartHeader` (line 215) exist specifically so a user-supplied `--filename` override containing a `"` or `\` cannot break out of the `Content-Disposition` header — the doc comment says so explicitly. Before this PR the stored name was always `filepath.Base(filePath)`, a value the OS already constrains; now it can be arbitrary user input via `--filename`, making this escaping the load-bearing safety net for a new, untrusted input path. Neither `attachments_test.go` (api) nor the cmd-level tests exercise an override containing a quote or backslash, so a regression in the escaping (e.g. reordering the `strings.NewReplacer` pairs, or dropping the call) would ship undetected. Add a case to `TestAddAttachment` with `override: `before"exploit.png`" (or similar) and assert the multipart reader still parses a single part with the expected literal filename, proving the header isn't corrupted.

*Reply inline to this comment.*

File: tools/jtk/internal/cmd/attachments/attachments_test.go [hunk 2]
```diff
@@ -293,12 +298,21 @@ func TestRunAdd_PartialFailure(t *testing.T) {
...
 	opts.SetAPIClient(client)
 
-	err = runAdd(context.Background(), opts, "TEST-1", []string{good, bad})
+	err = runAdd(context.Background(), opts, "TEST-1", []string{good, bad}, "")
 	testutil.RequireError(t, err)
 	testutil.Contains(t, stdout.String(), "good.txt")
 	testutil.Contains(t, stdout.String(), "10001")
 }
 
+func TestRunAdd_FilenameRequiresSingleFile(t *testing.T) {
```
Comment (piekstra-dev) [inline tools/jtk/internal/cmd/attachments/attachments_test.go:308]: `TestRunAdd_FilenameRequiresSingleFile` only proves the guard clause rejects `--filename` with multiple files; no test drives `runAdd` end-to-end with a single file plus `--filename` and asserts the server-observed filename matches the override. `AddAttachment(ctx, issueKey, filePath, filename string)` takes three same-typed string args, so a copy-paste/order regression in `runAdd`'s call (attachments.go:157) wouldn't be caught by the api-layer test (which calls `client.AddAttachment` directly) or by this guard-only test. Add a success-path test mirroring `TestRunAdd_Success` that passes a non-empty `filename` and asserts the httptest server received that name via `part.FileName()`.

*Reply inline to this comment.*

**Review by piekstra-dev — COMMENTED** [review]
## Automated PR Review

**Reviewed commit:** `029f0bae0488`
**Profile:** `reviewer` - **Posting as:** `piekstra-dev`

### Summary

| Reviewer | Findings |
|----------|----------|
| go:implementation-tests | 2 |
| policies:conventions | 0 |

<details>
<summary><strong>go:implementation-tests (2 findings)</strong></summary>

### Major - `tools/jtk/api/attachments.go:23`

> `quoteEscaper` and its use in `attachmentPartHeader` (line 215) exist specifically so a user-supplied `--filename` override containing a `"` or `\` cannot break out of the `Content-Disposition` header — the doc comment says so explicitly. Before this PR the stored name was always `filepath.Base(filePath)`, a value the OS already constrains; now it can be arbitrary user input via `--filename`, making this escaping the load-bearing safety net for a new, untrusted input path. Neither `attachments_test.go` (api) nor the cmd-level tests exercise an override containing a quote or backslash, so a regression in the escaping (e.g. reordering the `strings.NewReplacer` pairs, or dropping the call) would ship undetected. Add a case to `TestAddAttachment` with `override: `before"exploit.png`" (or similar) and assert the multipart reader still parses a single part with the expected literal filename, proving the header isn't corrupted.

### Minor - `tools/jtk/internal/cmd/attachments/attachments_test.go:307`

> `TestRunAdd_FilenameRequiresSingleFile` only proves the guard clause rejects `--filename` with multiple files; no test drives `runAdd` end-to-end with a single file plus `--filename` and asserts the server-observed filename matches the override. `AddAttachment(ctx, issueKey, filePath, filename string)` takes three same-typed string args, so a copy-paste/order regression in `runAdd`'s call (attachments.go:157) wouldn't be caught by the api-layer test (which calls `client.AddAttachment` directly) or by this guard-only test. Add a success-path test mirroring `TestRunAdd_Success` that passes a non-empty `filename` and asserts the httptest server received that name via `part.FileName()`.

</details>

### Reviewer Coverage

| Reviewer | Status | Inspected | Skipped | Constraints |
|----------|--------|-----------|---------|-------------|
| go:implementation-tests | complete_broad | tools/jtk/api/attachments.go, tools/jtk/api/attachments_test.go, tools/jtk/internal/cmd/attachments/attachments.go, tools/jtk/internal/cmd/attachments/attachments_test.go | unavailable | unavailable |
| policies:conventions | complete_broad | tools/jtk/README.md, tools/jtk/internal/cmd/attachments/attachments.go | unavailable | unavailable |

*0 PR discussion threads considered. 0 summarized; 0 resolved.*

---
<details>
<summary>Completed in 3m 16s | unavailable | claude-sonnet-5 | cr 0.10.264</summary>

| Field | Value |
|---|---|
| Model | claude-sonnet-5 |
| Reviewers | go:implementation-tests, policies:conventions |
| Engine | claude_cli · claude-sonnet-5 |
| Reviewed by | cr · piekstra-dev |
| Duration | 3m 16s wall · 4m 39s compute |
| Cost | unavailable |
| Tokens | unavailable |

**Per-workstream usage**

| Workstream | Model | In | Out | Cache read | Cache create | Cost | Duration |
|---|---|---:|---:|---:|---:|---:|---:|
| orchestrator-selection | claude-sonnet-5 | unavailable | unavailable | unavailable | unavailable | unavailable | 36s |
| go:implementation-tests | claude-sonnet-5 | unavailable | unavailable | unavailable | unavailable | unavailable | 2m 14s |
| policies:conventions | claude-sonnet-5 | unavailable | unavailable | unavailable | unavailable | unavailable | 1m 34s |
| orchestrator-rollup | claude-sonnet-5 | unavailable | unavailable | unavailable | unavailable | unavailable | 14s |

</details>

File: tools/jtk/api/attachments.go (hunk 1 above)
Comment (piekstra) [reply tools/jtk/api/attachments.go:23]: Good catch — `--filename` is the first untrusted value to reach that header, and the escaping was only asserted by the doc comment. Added quote and backslash override cases to `TestAddAttachment` (4733699): each asserts the filename round-trips to its literal value and that exactly one multipart part is produced, so dropping or reordering the `quoteEscaper` pairs now fails a test.

File: tools/jtk/internal/cmd/attachments/attachments_test.go (hunk 2 above)
Comment (piekstra) [reply tools/jtk/internal/cmd/attachments/attachments_test.go:308]: Added `TestRunAdd_FilenameOverride` (4733699): it drives `runAdd` end-to-end with a single file plus `--filename` and asserts the httptest server observes the overridden name via `part.FileName()`, covering the cmd-layer call site that the api-level test bypasses.

**Review by piekstra-dev — APPROVED** [review]
## Automated PR Review

**Reviewed commit:** `4733699c161f`
**Profile:** `reviewer` - **Posting as:** `piekstra-dev`

### Summary

| Reviewer | Findings |
|----------|----------|
| go:implementation-tests | 0 |
| policies:conventions | 0 |

### Reviewer Coverage

| Reviewer | Status | Inspected | Skipped | Constraints |
|----------|--------|-----------|---------|-------------|
| go:implementation-tests | complete_broad | tools/jtk/api/attachments.go, tools/jtk/api/attachments_test.go, tools/jtk/internal/cmd/attachments/attachments.go, tools/jtk/internal/cmd/attachments/attachments_test.go | unavailable | unavailable |
| policies:conventions | complete_broad | tools/jtk/README.md, tools/jtk/internal/cmd/attachments/attachments.go | unavailable | unavailable |

*0 PR discussion threads considered. 0 summarized; 0 resolved.*

---
<details>
<summary>Completed in 2m 39s | unavailable | claude-sonnet-5 | cr 0.10.264</summary>

| Field | Value |
|---|---|
| Model | claude-sonnet-5 |
| Reviewers | go:implementation-tests, policies:conventions |
| Engine | claude_cli · claude-sonnet-5 |
| Reviewed by | cr · piekstra-dev |
| Duration | 2m 39s wall · 3m 11s compute |
| Cost | unavailable |
| Tokens | unavailable |

**Per-workstream usage**

| Workstream | Model | In | Out | Cache read | Cache create | Cost | Duration |
|---|---|---:|---:|---:|---:|---:|---:|
| orchestrator-selection | claude-sonnet-5 | unavailable | unavailable | unavailable | unavailable | unavailable | 15s |
| go:implementation-tests | claude-sonnet-5 | unavailable | unavailable | unavailable | unavailable | unavailable | 1m 49s |
| policies:conventions | claude-sonnet-5 | unavailable | unavailable | unavailable | unavailable | unavailable | 55s |
| orchestrator-rollup | claude-sonnet-5 | unavailable | unavailable | unavailable | unavailable | unavailable | 11s |

</details>
